proposals/053-cryptography-extension.md

```diff
@@ -0,0 +1,124 @@
+
+- Stream ciphers
+- Complete X509 infrastructure implementation
+- Complete TLS server implementation
+
+One of the previous stretch goals was to develop the C bindings for X509 certificate architecture, and to [contribute that code back](https://github.com/randombit/botan/issues/3627) to `Botan C++`. Substantial progress was made, but further work is required in order to complete this. 
+
+Another feature missing from the Botan C FFI is Stream Ciphers, and like the X509 bindings, it will require effort to extend the Botan C++ library in order to expose it to Haskell.
+
+A third feature missing from the Botan C FFI is TLS, as Botan provides a full TLS server implementation, which will also require effort to expose to the FFI.
+
+## Development of cryptographic typeclasses and data families
+
+Due to the insufficiency of the ADT approach towards managing algorithms, cryptographic typeclasses and data families have been brought back into scope as a deliverable after being declared out-of-scope in the original proposal.
+
+While a first attempt has already been made to typeclassify the various cryptographic primitives, this interface will require extensive refinement and feedback over the coming months, and is likely to experience significant change before settling down.
+
+It is expected that these typeclasses will result in improved per-algorithm type safety, and better ergonomics through more consistent handling of cryptographic primitives such as keys, nonces, and ciphertexts.
+
+## Improvements to installation / CI / Unit tests
+
+One of the largest warts in using these libraries is the installation of the Botan C++ library. Package distributions are available for some users and operating systems via `pkg-config`, but other users currently require manual installation.
+
+This is a one-time thing, but it still is a significant pain point in what should otherwise be a much smoother process. As such, research into packaging or bundling Botan C++ is ongoing, with multiple potential solutions already having been identified.
+
+This is a high-priority goal, alongside improvements to CI and unit testing, and will be a major focus.
+
+# Goals
+
+The following are the immediate goals / deliverables of this proposal:
+
+    - To continue to refine the `botan` libraries
+    - To continue improvements to the Botan C FFI and Haskell bindings
+    - To develop an abstract cryptographic typeclass / data family hierarchy exhibiting per-algorithm safety
+    - To improve Botan installation / CI ease of use
+
+The following are intermediate deliverables:
+
+    - Frequent activity update on devlog
+    - Weekly major update that includes github push
+    - Monthly report that summarizes what specific items have been or will be worked on, and what challenges have arisen, if any.
+
+The following are stretch goals / optional deliverables to be looked into only after required deliverables are met:
```

Comment:
How many stretch goals could be complete, assuming an optimistic scenario? I feel like 5 goals are beyond possible even if everything goes very smooth, so I'd suggest to whittle down the list to a more realistic shape.